ext/standard/http_fopen_wrapper.c

```diff
@@ -352,6 +352,43 @@ static zend_string *php_stream_http_response_headers_parse(php_stream_wrapper *w
 	return NULL;
 }
 
+static bool php_stream_unwrap_content(php_stream_context *context, zend_string **str_out, php_stream **stream_out)
+{
+	zval *content = php_stream_context_get_option(context, "http", "content");
+	if (content) {
+		if (Z_TYPE_P(content) == IS_STRING && Z_STRLEN_P(content) > 0) {
+			*str_out = Z_STR_P(content);
+			return true;
+		} else if (Z_TYPE_P(content) == IS_RESOURCE) {
+			if ((php_stream_from_zval_no_verify(*stream_out, content))) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+static bool php_stream_append_content_length(smart_str *req_buf, zend_string *content_str, php_stream *content_stream)
+{
+	smart_str_appends(req_buf, "Content-Length: ");
+	if (content_str) {
+		smart_str_append_unsigned(req_buf, ZSTR_LEN(content_str));
+	} else {
+		zend_off_t current_position = php_stream_tell(content_stream);
+		if (php_stream_seek(content_stream, 0, SEEK_END) < 0) {
+			return false;
+		}
+		zend_off_t end_position = php_stream_tell(content_stream);
+		if (php_stream_seek(content_stream, current_position, SEEK_SET) < 0) {
+			return false;
+		}
+		smart_str_append_unsigned(req_buf, end_position - current_position);
```

Comment:
What about the non seekable streams? 

I'm thinking that it might be better to use chunked encoding for such cases

Reply:
I'll go with the "read to memory" solution as I believe that non-seekable streams are pretty rare in this context.